test/release/examples/users-guide/datapar/forall.chpl

```diff
@@ -0,0 +1,49 @@
+config param runSerial = false,
+             runConcurrent = false,
+             runRace = false;;
+
+config 
+const n = 1000000;
+var A: [1..n] real;
+
+if runSerial {
+
+for a in A do
+  a += 1.0;
+
+} else if runConcurrent {
+
+coforall a in A do
+  a += 1.0;
+
+} else {
+
+forall a in A do
+  a += 1.0;
+
+}
+
+writeln(A);
+writeln();
+
+if runRace {
+
+forall i in 2..n-1 do
+  A[i] = A[i-1] + A[i+1];
+writeln(A);
+writeln();
+
+} else {
+
+forall i in 2..n-1 by 2 do
+  A[i] = A[i-1] + A[i+1];
```

Comment:
This example doesn't seem like a race condition, as the odd indices are being read while the even indices are being written. If the "by 2" wasn't there then it would definitely be a race.